### PR TITLE
[Restoration Shaman] Flash Floods update to talent ranks

### PR DESCRIPTION
**Pull request by avilene**

### CODE REVIEW

File: src/analysis/retail/shaman/restoration/modules/talents/FlashFlood.tsx [hunk 1]
```diff
@@ -63,6 +63,8 @@ class FlashFlood extends Analyzer {
   constructor(options: Options) {
     super(options);
     this.active = this.selectedCombatant.hasTalent(TALENTS.FLASH_FLOOD_TALENT.id);
+    this.flashFloodHaste =
+      this.selectedCombatant.getTalentRank(TALENTS.FLASH_FLOOD_TALENT.id) * 0.1;
```
Comment (danny-janse) [inline src/analysis/retail/shaman/restoration/modules/talents/FlashFlood.tsx]: I would recommend turning the 0.1 into a constant instead. This way there will be no question in the future what this multiplication is for.

File: src/analysis/retail/shaman/restoration/modules/talents/FlashFlood.tsx [hunk 2]
```diff
@@ -63,6 +63,8 @@ class FlashFlood extends Analyzer {
   constructor(options: Options) {
     super(options);
     this.active = this.selectedCombatant.hasTalent(TALENTS.FLASH_FLOOD_TALENT.id);
```
Comment (danny-janse) [inline src/analysis/retail/shaman/restoration/modules/talents/FlashFlood.tsx:65]: I'd recommend adding a return if the module should not be active before adding listeners and doing other logic.

**Review by danny-janse — CHANGES_REQUESTED** [review]
Looks good! Could you make the two requested changed and add a changelog to finish it up?

File: src/analysis/retail/shaman/restoration/modules/talents/FlashFlood.tsx (hunk 2 above)
Comment (kfinch) [reply src/analysis/retail/shaman/restoration/modules/talents/FlashFlood.tsx:65]: I've been specifically suggesting not to do this. It's hidden behavior, but listeners actually don't get added for inactive modules.

Comment (danny-janse) [reply src/analysis/retail/shaman/restoration/modules/talents/FlashFlood.tsx:65]: Oh interesting. I'm aware that listeners are not executed. The reason I've been suggesting this isbecause there can be unused code still running. Like calculating the effectiveness of an ability. Eventho its not a listener it is still code that has to be exectued in the browser - aka the end-user.

Comment (kfinch) [reply src/analysis/retail/shaman/restoration/modules/talents/FlashFlood.tsx:65]: It's such a trivial amount of extra code I doubt it matters much, but point taken. I still think the no return looks cleaner 🤷

**Review by kfinch — APPROVED** [review]

**Review by danny-janse — APPROVED** [review]

**Pull request merged**